flocker/node/agents/blockdevice.py

```diff
@@ -551,29 +551,18 @@ def run(self, deployer):
         device.  The volume must be attached to this node and the corresponding
         block device mounted.
         """
-        api = deployer.async_block_device_api
-        listing = api.list_volumes()
-        listing.addCallback(
-            _blockdevice_volume_from_datasetid, self.dataset_id
-        )
+        try:
+            device = get_device_for_dataset_id(self.dataset_id)
+        except KeyError:
+            # It was not actually found.
+            return fail(DatasetWithoutVolume(dataset_id=self.dataset_id))
```

Comment:
Mmmm. I feel like it's not quite the same, the former is more like "pretty sure this is not a thing in the whole system" the latter is "sorry couldn't find it locally".